cmd/mount_zfs/mount_zfs.c

```diff
@@ -47,63 +47,44 @@ libzfs_handle_t *g_zfs;
  * is expected to be of the form pool/dataset, however may also refer to
  * a block device if that device contains a valid zfs label.
  */
-static char *
-parse_dataset(char *dataset)
+static void
+parse_dataset(char (*dataset)[PATH_MAX])
 {
-	char cwd[PATH_MAX];
-	struct stat64 statbuf;
-	int error;
-	int len;
-
 	/*
 	 * We expect a pool/dataset to be provided, however if we're
 	 * given a device which is a member of a zpool we attempt to
 	 * extract the pool name stored in the label.  Given the pool
 	 * name we can mount the root dataset.
 	 */
-	error = stat64(dataset, &statbuf);
-	if (error == 0) {
+	int fd = open((const char *)dataset, O_RDONLY);
+	if (fd > 0) {
```

Comment:
Valid fds start from 0. While this probably will happen to not matter the condition should not look like that.